### PR TITLE
sidekick 0.5.3 (new formula)

### PR DESCRIPTION
**Pull request by bevanjkay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Suggesting `sidekick-cli` for the name, because there is an overlap with an existing cask `sidekick`.

### CODE REVIEW

File: Formula/s/sidekick-cli.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  desc "Deploy applications to your VPS"
+  homepage "https://github.com/MightyMoud/sidekick"
+  url "https://github.com/MightyMoud/sidekick/archive/refs/tags/v0.5.3.tar.gz"
+  sha256 "11c6f23c67c122ec6e1e0fa4cd7119f7e044186abe810e08ea97708bb78e0a33"
+  license "GPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
```
Comment (chenrui333) [inline Formula/s/sidekick-cli.rb:11]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w", output: bin/"sidekick")
```

**Comment by chenrui333** [comment]
> Suggesting `sidekick-cli` for the name, because there is an overlap with an existing cask `sidekick`.

I dont feel like we need to do it because of the name overlapping?

File: Formula/s/sidekick-cli.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    bin.install bin/"sidekick-cli" => "sidekick"
+  end
+
+  test do
+    assert_match "With sidekick you can deploy any number of applications to a single VPS",
+                  shell_output(bin/"sidekick")
+    assert_match(/Sidekick config not found - Run sidekick init/, shell_output(bin/"sidekick compose", 1))
```
Comment (carlocab) [inline Formula/s/sidekick-cli.rb:18]:
```suggestion
    assert_match(/Sidekick config not found - Run sidekick init/, shell_output("#{bin}/sidekick compose", 1))
```
`bin/"sidekick compose"` is not a `Pathname`. (But `bin/"sidekick"` is.)

**Comment by bevanjkay** [comment]
> I dont feel like we need to do it because of the name overlapping?

@chenrui333  Was based on the comment here - https://github.com/Homebrew/brew/issues/16309#issuecomment-2128940456 - which isn't really documented. But I'm not too concerned either way.

File: Formula/s/sidekick-cli.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w", output: bin/"sidekick")
+  end
+
+  test do
+    assert_match "With sidekick you can deploy any number of applications to a single VPS",
+                  shell_output(bin/"sidekick")
+    assert_match(/Sidekick config not found - Run sidekick init/, shell_output("#{bin}/sidekick compose", 1))
```
Comment (chenrui333) [inline Formula/s/sidekick-cli.rb:17]:
```suggestion
    assert_match "Sidekick config not found - Run sidekick init", shell_output("#{bin}/sidekick compose", 1)
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
> > I dont feel like we need to do it because of the name overlapping?
> 
> @chenrui333 Was based on the comment here - [Homebrew/brew#16309 (comment)](https://github.com/Homebrew/brew/issues/16309#issuecomment-2128940456) - which isn't really documented. But I'm not too concerned either way.

I see, sounds good then :)

**Review by carlocab — APPROVED** [review]

File: Formula/s/sidekick-cli.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class SidekickCli < Formula
```
Comment (carlocab) [inline Formula/s/sidekick-cli.rb:1]: I think we can just call this `sidekick` -- the Sidekick cask is discontinued.

**Review by carlocab — APPROVED** [review]

**Pull request merged**